File: android-rider-app/app/src/main/java/com/meponto/rider/ui/theme/Theme.kt
package com.meponto.rider.ui.theme

import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Shapes
import androidx.compose.material3.darkColorScheme
import androidx.compose.material3.lightColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.runtime.CompositionLocalProvider
import androidx.compose.runtime.staticCompositionLocalOf
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp

/**
 * MePonto design tokens, mirrored from docs/design-system.md and app/globals.css
 * (and kept 1:1 with the iOS rider app's Theme.swift). Dark is the default
 * palette; a light palette backs the system/light toggle. Components must always
 * reference these semantic tokens, never raw hex.
 */
data class MeColors(
    val background: Color,
    val surface: Color,
    val surfaceRaised: Color,
    val surfaceHover: Color,
    val line: Color,
    val text: Color,
    val textSoft: Color,
    val muted: Color,
    val accent: Color,
    val accentInk: Color,
    val danger: Color,
    val warning: Color,
    val ok: Color,
    val isDark: Boolean,
)

// Consumer-grade dual palette (delivery-app vibe): light is the default and
// carries saturated brand yellow + lively functional colors; dark is a richer
// nightly variant of the same identity.
val DarkMeColors = MeColors(
    background = Color(0xFF0B0C10),
    surface = Color(0xFF15171E),
    surfaceRaised = Color(0xFF1C1F28),
    surfaceHover = Color(0xFF242836),
    line = Color(0xFF2C3040),
    text = Color(0xFFF6F7FB),
    textSoft = Color(0xFFD8DCE6),
    muted = Color(0xFF98A0AF),
    accent = Color(0xFFFFD100),
    accentInk = Color(0xFF1A1500),
    danger = Color(0xFFFF5C70),
    warning = Color(0xFFFFB454),
    ok = Color(0xFF2DE0A5),
    isDark = true,
)

val LightMeColors = MeColors(
    background = Color(0xFFF6F6F8),
    surface = Color(0xFFFFFFFF),
    surfaceRaised = Color(0xFFF2F3F7),
    surfaceHover = Color(0xFFECEEF4),
    line = Color(0xFFE4E7EE),
    text = Color(0xFF17181C),
    textSoft = Color(0xFF3C4148),
    muted = Color(0xFF7A8291),
    accent = Color(0xFFFFC400),
    accentInk = Color(0xFF201A00),
    danger = Color(0xFFFF4D5E),
    warning = Color(0xFFFF9F1C),
    ok = Color(0xFF00B884),
    isDark = false,
)

val LocalMe = staticCompositionLocalOf { DarkMeColors }

/** Status tone used by badges and ledgers. */
enum class Tone { NEUTRAL, ACCENT, OK, WARNING, DANGER }

fun Tone.fg(c: MeColors): Color = when (this) {
    Tone.NEUTRAL -> c.textSoft
    Tone.ACCENT -> c.accent
    Tone.OK -> c.ok
    Tone.WARNING -> c.warning
    Tone.DANGER -> c.danger
}

fun Tone.bg(c: MeColors): Color = when (this) {
    Tone.NEUTRAL -> c.surfaceRaised
    else -> fg(c).copy(alpha = 0.14f)
}

object MeRadius {
    val card = 16.dp
    val small = 10.dp
}

@Composable
fun MePontoTheme(darkTheme: Boolean, content: @Composable () -> Unit) {
    val me = if (darkTheme) DarkMeColors else LightMeColors
    val colorScheme = if (darkTheme) {
        darkColorScheme(
            primary = me.accent,
            onPrimary = me.accentInk,
            background = me.background,
            surface = me.surface,
            surfaceVariant = me.surfaceRaised,
            outline = me.line,
            onBackground = me.text,
            onSurface = me.text,
        )
    } else {
        lightColorScheme(
            primary = me.accent,
            onPrimary = me.accentInk,
            background = me.background,
            surface = me.surface,
            surfaceVariant = me.surfaceRaised,
            outline = me.line,
            onBackground = me.text,
            onSurface = me.text,
        )
    }
    // Rounded shape scale: text fields, sheets and menus pick these up.
    val shapes = Shapes(
        extraSmall = RoundedCornerShape(10.dp),
        small = RoundedCornerShape(12.dp),
        medium = RoundedCornerShape(16.dp),
        large = RoundedCornerShape(20.dp),
        extraLarge = RoundedCornerShape(28.dp),
    )
    CompositionLocalProvider(LocalMe provides me) {
        MaterialTheme(colorScheme = colorScheme, shapes = shapes, content = content)
    }
}

File: ios-rider-app/MePontoRider/Theme/Theme.swift
import SwiftUI

// MePonto design tokens, mirrored from docs/design-system.md and app/globals.css.
// Dark is the default palette; a light palette is provided for the system toggle.
// Components must reference these semantic tokens, never raw hex.
enum Theme {

    // MARK: - Hex helper
    static func color(_ hex: UInt32, alpha: Double = 1) -> Color {
        let r = Double((hex >> 16) & 0xFF) / 255.0
        let g = Double((hex >> 8) & 0xFF) / 255.0
        let b = Double(hex & 0xFF) / 255.0
        return Color(.sRGB, red: r, green: g, blue: b, opacity: alpha)
    }

    // MARK: - Semantic tokens (dynamic: resolve per color scheme)
    static func background(_ s: ColorScheme) -> Color { s == .dark ? color(0x07090d) : color(0xf5f7fa) }
    static func surface(_ s: ColorScheme) -> Color { s == .dark ? color(0x0d1117) : color(0xffffff) }
    static func surfaceRaised(_ s: ColorScheme) -> Color { s == .dark ? color(0x111722) : color(0xf8fafc) }
    static func surfaceHover(_ s: ColorScheme) -> Color { s == .dark ? color(0x172031) : color(0xeef3f8) }
    static func line(_ s: ColorScheme) -> Color { s == .dark ? color(0x263244) : color(0xd8e0ea) }
    static func text(_ s: ColorScheme) -> Color { s == .dark ? color(0xf8fafc) : color(0x111827) }
    static func textSoft(_ s: ColorScheme) -> Color { s == .dark ? color(0xd7dee8) : color(0x334155) }
    static func muted(_ s: ColorScheme) -> Color { s == .dark ? color(0x9aa6b8) : color(0x64748b) }
    static func accent(_ s: ColorScheme) -> Color { s == .dark ? color(0xffd100) : color(0xd9a900) }
    static func accentInk(_ s: ColorScheme) -> Color { color(0x171400) }
    static func danger(_ s: ColorScheme) -> Color { s == .dark ? color(0xff5c70) : color(0xdc2626) }
    static func warning(_ s: ColorScheme) -> Color { s == .dark ? color(0xffb454) : color(0xb45309) }
    static func ok(_ s: ColorScheme) -> Color { s == .dark ? color(0x2dd4bf) : color(0x0f766e) }

    static let radius: CGFloat = 8
    static let radiusSmall: CGFloat = 6
}

// Status tone used by Badge and ledgers.
enum Tone {
    case neutral, accent, ok, warning, danger

    func fg(_ s: ColorScheme) -> Color {
        switch self {
        case .neutral: return Theme.textSoft(s)
        case .accent: return Theme.accent(s)
        case .ok: return Theme.ok(s)
        case .warning: return Theme.warning(s)
        case .danger: return Theme.danger(s)
        }
    }

    func bg(_ s: ColorScheme) -> Color {
        switch self {
        case .neutral: return Theme.surfaceRaised(s)
        default: return fg(s).opacity(0.14)
        }
    }
}

// Convenience environment-aware accessor for views.
struct ThemeColors {
    let scheme: ColorScheme
    var background: Color { Theme.background(scheme) }
    var surface: Color { Theme.surface(scheme) }
    var surfaceRaised: Color { Theme.surfaceRaised(scheme) }
    var surfaceHover: Color { Theme.surfaceHover(scheme) }
    var line: Color { Theme.line(scheme) }
    var text: Color { Theme.text(scheme) }
    var textSoft: Color { Theme.textSoft(scheme) }
    var muted: Color { Theme.muted(scheme) }
    var accent: Color { Theme.accent(scheme) }
    var accentInk: Color { Theme.accentInk(scheme) }
    var danger: Color { Theme.danger(scheme) }
    var warning: Color { Theme.warning(scheme) }
    var ok: Color { Theme.ok(scheme) }
}

extension EnvironmentValues {
    var t: ThemeColors { ThemeColors(scheme: self.colorScheme) }
}
